File: frontend/src/types/report.ts
// frontend/src/types/report.ts

export interface ReportEdge {
  source: string;
  target: string;
  rank: number;
  score_drop: number;
}

export interface ReportReview {
  edge_source: string;
  edge_target: string;
  status: string;
  memo: string;
}

export interface ReportContent {
  analysis_id: string;
  dataset: string;
  target_transaction: string;
  model_name: string;
  model_version: string;
  prediction_score: number;
  prediction_level: string;
  recommended_edges: ReportEdge[];
  reviews: ReportReview[];
}

export interface ReportResponse {
  report_id: string;
  content: ReportContent;
  generated_at: string;
  report_hash: string;
}

File: backend/app/models/report.py
"""backend/app/models/report.py
Pydantic models for Investigation Reports.
"""
from typing import List, Any
from pydantic import BaseModel
from datetime import datetime

class ReportCreateRequest(BaseModel):
    analysis_id: str
    reviews: List['ReportReview']

class ReportEdge(BaseModel):
    source: str
    target: str
    rank: int
    score_drop: float

class ReportReview(BaseModel):
    edge_source: str
    edge_target: str
    status: str
    memo: str

class ReportContent(BaseModel):
    analysis_id: str
    dataset: str
    target_transaction: str
    model_name: str
    model_version: str
    prediction_score: float
    prediction_level: str
    recommended_edges: List[ReportEdge]
    reviews: List[ReportReview]

    model_config = {"protected_namespaces": ()}

class ReportResponse(BaseModel):
    report_id: str
    content: ReportContent
    generated_at: str
    report_hash: str
